openwisp_controller/subnet_division/tests/test_models.py

```diff
@@ -246,6 +250,73 @@ def test_field_validations(self):
                 {'organization': ['Organization should be same as the subnet']},
             )
 
+    def test_slash_32_rule_ipv4(self):
+        rule = self._get_vpn_subdivision_rule(
+            size=32, number_of_ips=1, number_of_subnets=1
+        )
+        self.config.templates.add(self.template)
+        rule.subnetdivisionindex_set.count()
+        index_queryset = rule.subnetdivisionindex_set.filter(
+            config_id=self.config.id, subnet_id__isnull=False, ip_id__isnull=False
+        )
+        self.assertEqual(index_queryset.count(), 1)
+        index = index_queryset.first()
+        self.assertEqual(str(index.subnet.subnet), '10.0.0.1/32')
+        self.assertEqual(index.ip.ip_address, '10.0.0.1')
+
+    def test_slash_32_rule_ipv4_error(self):
```

Comment:
Here I try to create a subnet division rule for a /32 master subnet, which cannot work because it only has 1 slot available, which would be taken by the reserved subnet and the code would fail. To avoid that I made sure the system prevent from saving such a rule.